lib/phubme/nicknames_matcher.ex

```diff
@@ -1,31 +1,27 @@
 defmodule PhubMe.NicknamesMatcher do
   def match_nicknames({full_comment, github_nicknames, sender, comment_parsed}) do
-    {:ok, full_comment, matching_nicknames(github_nicknames, [], 0), sender, comment_parsed }
+    {:ok, full_comment, matching_nicknames(github_nicknames), sender, comment_parsed }
   end
 
   def match_nicknames(_) do
     {:no_nicknames_found}
   end
 
-  defp matching_nicknames([github_nick_head | github_nick_tail], matched_nicknames, array_position) do
-    updated_matched_nicknames =
-      if nickname_from_mix_config(github_nick_head) == :error do
```

Comment:
You were calling this `nickname_from_mix_config(github_nick_head)` twice. In this commit, the `case` statement is keeping the interesting part of the result for us.